app/controllers/missions_controller.rb

```diff
@@ -39,50 +39,23 @@ def update
 
   def destroy
     if @mission.destroy
-      redirect_to root_path, notice: t('.success')
+      redirect_to missions_path, notice: t('.success')
     else
-      redirect_to root_path, notice: t('.failure')
+      redirect_to missions_path, notice: t('.failure')
     end
   end
 
   private
 
+  def authenticate_user!
+    redirect_to login_path, notice: t('missions.failure') unless session[:current_user_id]
+  end
```

Comment:
Session 中有 user id 並不一定代表真的有 user 真的還存在（例：某 user 還在登入時，該 user record 在 Rails console 被刪除），請確保實際上真的有 user 存在。
```suggestion
  def authenticate_user!
    redirect_to login_path, notice: t('missions.failure') if current_user.blank?
  end
```

Reply:
@Dantecheng41